compliance_checker/imos/imos.py

```diff
@@ -1311,6 +1311,29 @@ def check_quality_control_conventions_for_quality_control_variable(self, dataset
 
         return ret_val
 
+    def check_quality_control_variable_matches_variable(self, dataset):
+        """
+        Check that the name of a quality control variable which is <DATA>_quality_control
+        matches the name of a corresponding data variable <DATA>.
+        """
+        ret_val = []
+        for qc_variable in self._quality_control_variables:
+            qc_variable_root_name = qc_variable.name[:-16] # '_quality_control' is 16 characters long
```

Comment:
Isn't checking this redundant? 

```self._quality_control_variables``` already returns the list of variables ending with '_quality_control'. Maybe I could test that this list is not empty before continuing.